adaptdl/adaptdl/goodput.py

```diff
@@ -124,8 +123,19 @@ def optimize(self, num_nodes, num_replicas, max_batch_size=None,
         goodput = self.evaluate(num_nodes, num_replicas,
                                 atomic_bsz, accum_steps)
         # Set the goodput of invalid configurations to 0.0.
-        goodput = np.where((min_atomic_bsz <= atomic_bsz) &
-                           (atomic_bsz <= max_atomic_bsz), goodput, 0.0)
+        goodput = np.where(
+            np.logical_and(
+                min_atomic_bsz <= atomic_bsz,
+                np.logical_and(
+                    atomic_bsz <= max_atomic_bsz,
+                    np.logical_or(
+                        # Disallow 1 replica scaling without
+                        # gradient accumulation
+                        num_replicas > 1,
+                        np.logical_or(
+                            accum_steps > 0,
+                            atomic_bsz == self._init_batch_size)))),
+            goodput, 0.0)
```

Comment:
What's the context for this change?

Reply:
So right now this code allows us to pick larger batchsizes than the initial/default while on a single replica. I think we decided to disallow this a while back. This change re-disables it.